src/util/imagefiledata.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <QByteArray>
+#include <QFile>
+#include <QImage>
+
+class ImageFileData : public QImage {
+  public:
+    explicit ImageFileData(const QByteArray& coverArtBytes, const char* format = nullptr);
```

Comment:
After the latest changes a format is mandatory. This can become a QByteArray, to get around the "\0" check for the string end.

Reply:
The default to nullptr is misleading, because we need the format in this implementation. 

We may end 
```fromByteArray(const QByteArray& data);```
for the case we have only the data but do not know the format yet.

Reply:
Wait, we can make the format optional in the constructor. Than we need to lookup the format from the QByteArray like in fromFilePath()

Reply:
I have tried to do it with a new static method getFormat(), I am not sure if that should've been in anon namespace or not.

Also, is there a way to get a cover art without format specified? Should we also take care of this circumstance? Such as hard coded "jpg" format?

Reply:
A static function is OK. However getFormat() sound like a member that return the format. Maybe readFormat() or such?

Reply:
It actually returns the format, but still readFormat is also well explanatory